hibernate-core/src/main/java/org/hibernate/metamodel/internal/MetamodelImpl.java

```diff
@@ -623,12 +625,20 @@ else if ( result == INVALID_IMPORT ) {
 	 */
 	public String[] getImplementors(String className) throws MappingException {
 
+		if (implementorsCache.get(className) != null) {
+			return implementorsCache.get(className);
+		}
+
 		final Class clazz;
+		String[] implementors;
```

Comment:
Minor nitpick, but considering the flow, I would only initialize the variable where needed.